### PR TITLE
fix for login form invisible in blog page

### LINKED ISSUE #1847: Login form not appearing in blog(including all section in it like overview,team and all other sections) and not found page
**Actual Behaviour**
when trying to login in blog page form is appearing

**Expected Behaviour**
login form should appear

**Steps to reproduce it**
open [https://chat.susi.ai/blog](https://chat.susi.ai/blog) and try to login

**Screenshots of the issue**

login form not appering

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by prateek76**
Fixes #1847

Changes: now login form is visible in blog page on clicking on login

Demo Link: https://pr-1848-fossasia-susi-web-chat.surge.sh

Screenshots for the change: 
pull2

### CODE REVIEW

**Comment by prateek76** [comment]
@AakashMallik please review

**Comment by AakashMallik** [comment]
@prateek76 Please use the Login, Signup, and forgot password as they have been used in Topbar. There is a top-level function declared to dispatch the login, signup and forgot password dialog. Use that instead.

**Comment by prateek76** [comment]
> @prateek76 Please use the Login, Signup, and forgot password as they have been used in Topbar. There is a top-level function declared to dispatch the login, signup and forgot password dialog. Use that instead.

i don't get what you said can you explain @AakashMallik

**Comment by AakashMallik** [comment]
> > @prateek76 Please use the Login, Signup, and forgot password as they have been used in Topbar. There is a top-level function declared to dispatch the login, signup and forgot password dialog. Use that instead.
> 
> i don't get what you said can you explain @AakashMallik,but it's working fine

We don't just want **working** code, we want **elegant** code. :innocent:
See how we have called Login in **TopBar.react.js** file and try to implement it for **StaticAppBar** in a similar fashion.

**Comment by prateek76** [comment]
@AakashMallik i used the same way used in home page for login, what is the problem with the code?

**Comment by AakashMallik** [comment]
@prateek76 

```
<Route
              exact
              path="/"
              render={routeProps => (
                <ChatApp
                  {...routeProps}
                  onRequestOpenLogin={this.onRequestOpenLogin}
                />
              )}
            />
```

This is how you are supposed to pass the function to Blog component and then call this function when user click on the login button on static app bar. This will automatically open Login dialog I have declared in App.js. We are trying to maximize code reuse. Try to abide by that idea.

**Comment by prateek76** [comment]
screenshot from 2018-12-15 07-04-21
its giving result like this @AakashMallik

**Comment by AakashMallik** [comment]
@prateek you are still doing it wrong. Please try to understand how the code is working before making changes.

If you need further help, DM. But I would insist that you exhaust all your options before taking my help. :grinning:

**Comment by prateek76** [comment]
@AakashMallik @Akshat-Jain @akshatnitd please review

File: src/components/StaticAppBar/StaticAppBar.react.js [hunk 1]
```diff
@@ -551,54 +557,23 @@ class StaticAppBar extends Component {
...
-          onRequestClose={this.handleClose}
-        >
-          <ForgotPassword
-            {...this.props}
-            showForgotPassword={this.showForgotPassword}
-            onLoginSignUp={this.handleLogin}
-          />
-          <Close style={closingStyle} onTouchTap={this.handleClose} />
-        </Dialog>
+        <SignUp
```
Comment (AakashMallik) [inline src/components/StaticAppBar/StaticAppBar.react.js]: Not needed

File: src/components/StaticAppBar/StaticAppBar.react.js [hunk 2]
```diff
@@ -551,54 +557,23 @@ class StaticAppBar extends Component {
...
-          onRequestClose={this.handleClose}
-        >
-          <Login
-            {...this.props}
-            handleSignUp={this.handleSignUp}
-            handleForgotPassword={this.handleForgotPassword}
-          />
-          <Close style={closingStyle} onTouchTap={this.handleClose} />
-        </Dialog>
+        <Login
```
Comment (AakashMallik) [inline src/components/StaticAppBar/StaticAppBar.react.js]: Not needed

File: src/components/StaticAppBar/StaticAppBar.react.js [hunk 3]
```diff
@@ -193,6 +191,14 @@ class StaticAppBar extends Component {
     });
   };
 
+  onRequestClose = () => {
```
Comment (AakashMallik) [inline src/components/StaticAppBar/StaticAppBar.react.js]: Not needed

**Comment by AakashMallik** [comment]
Please pass the functions from App.js down to the static bar. All you need to do here is call those functions. You need not declare any component here.

**Comment by garvitkhandelwal** [comment]
@prateek76  Any updates? Can I work on this?

**Comment by prateek76** [comment]
> @prateek76 Any updates? Can I work on this?

i am woking on it

**Comment by AakashMallik** [comment]
Any progress?

**Comment by prateek76** [comment]
@AakashMallik i have passed the function to props but its not working at the moment trying to fix it

**Comment by AakashMallik** [comment]
@prateek76 Great that you are still active. It would be really nice if you could rush this a bit.

**Comment by mariobehling** [comment]
Closing this due to inactivity. Please reopen, if you interested to finalize this issue. Thank you.

**Pull request closed without merging**